67_RayQueryGeometry/app_resources/render.comp.hlsl

```diff
@@ -19,6 +19,16 @@ cbuffer CameraData
 
 [[vk::binding(2, 0)]] RWTexture2D<float4> outImage;
 
+// TODO: don't know if this is correct
+float3 getNormalsFromMask(int bn, int mask)
+{
+    float3 n = (float3)0;
+    n.x = float(asint((bn >> 24) & mask));
+    n.y = float((bn >> 16) & mask);
+    n.z = float((bn >> 8) & mask);
+    return n;
+}
```

Comment:
what are you trying to do?

Reply:
Read an int8_t3 from the buffer as an int32_t. So what I think needs doing is shift it down and read 8 bits as a float. I forgot to `asint` the other lines but don't think it's necessary

Reply:
read a `uint32_t` and then use the GLSL.450 extended intrinsic `unpack4x8Unorm`, implement it here: 
https://github.com/Devsh-Graphics-Programming/Nabla/blob/d37770f6dfab7f19de6c798ddd2c5a93256ee7ae/include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl#L198